rs/src/gstd/syscalls.rs

```diff
@@ -0,0 +1,215 @@
+    pub fn message_id() -> MessageId {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn message_size() -> usize {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn message_source() -> ActorId {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn message_value() -> u128 {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn program_id() -> ActorId {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn block_height() -> u32 {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn block_timestamp() -> u64 {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn value_available() -> u128 {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn env_vars() -> gstd::EnvVars {
+        unimplemented!("{ERROR}")
+    }
+
+    pub fn exit(_inheritor_id: ActorId) -> ! {
+        unimplemented!("{ERROR}")
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+#[cfg(feature = "std")]
+const _: () = {
+    use core::cell::RefCell;
+    use std::thread_local;
+
+    #[derive(Default, Clone, Copy)]
+    struct SyscallState {
+        message_id: MessageId,
+        message_size: usize,
+        message_source: ActorId,
+        message_value: u128,
+        program_id: ActorId,
+        block_height: u32,
+        block_timestamp: u64,
+        value_available: u128,
+    }
+
+    thread_local! {
+        static SYSCALL_STATE: RefCell<SyscallState> = RefCell::new(SyscallState::default());
+    }
+
+    impl Syscall {
+        pub fn message_id() -> MessageId {
+            SYSCALL_STATE.with_borrow(|state| state.message_id)
+        }
+
+        pub fn message_size() -> usize {
+            SYSCALL_STATE.with_borrow(|state| state.message_size)
+        }
+
+        pub fn message_source() -> ActorId {
+            SYSCALL_STATE.with_borrow(|state| state.message_source)
+        }
+
+        pub fn message_value() -> u128 {
+            SYSCALL_STATE.with_borrow(|state| state.message_value)
+        }
+
+        pub fn program_id() -> ActorId {
+            SYSCALL_STATE.with_borrow(|state| state.program_id)
+        }
+
+        pub fn block_height() -> u32 {
+            SYSCALL_STATE.with_borrow(|state| state.block_height)
+        }
+
+        pub fn block_timestamp() -> u64 {
+            SYSCALL_STATE.with_borrow(|state| state.block_timestamp)
+        }
+
+        pub fn value_available() -> u128 {
+            SYSCALL_STATE.with_borrow(|state| state.value_available)
+        }
+
+        pub fn env_vars() -> gstd::EnvVars {
+            gstd::EnvVars {
+                performance_multiplier: gstd::Percent::new(100),
+                existential_deposit: 1_000_000_000_000,
+                mailbox_threshold: 3000,
+                gas_multiplier: gstd::GasMultiplier::from_value_per_gas(100),
+            }
+        }
+
+        pub fn exit(inheritor_id: ActorId) -> ! {
+            panic!("Program exited with inheritor id: {}", inheritor_id);
+        }
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    #[cfg(feature = "std")]
```

Comment:
seems no need for that, as the whole block is marked with the same feature attributes, isn't it?